src/overview/reducer.js

```diff
@@ -161,6 +161,51 @@ const initSearchCount = (state, searchCount) => ({ ...state, searchCount })
 
 const payloadReducer = key => (state, payload) => ({ ...state, [key]: payload })
 
+// TODO: Think of some good variable names
+const bulkSetHasBookmark = (state, urls) => {
+    let { docs } = state.searchResult
+    const { hasBookmark } = docs[0]
+    const urlIndices = []
+    const sameBookmarks = docs.filter((doc, i) => {
+        if (urls.indexOf(doc.url) > -1) {
+            urlIndices.push(i)
+            return doc.hasBookmark !== hasBookmark
+        }
+    })
+
+    if (sameBookmarks.length > 0) {
+        urlIndices.map(index => {
```

Comment:
I don't think this reducer is being triggered from anywhere, as we're putting the bulk stuff on hold IIRC, but just in general: `Array.prototype.map` generally shouldn't be used for side-effects. Use `Array.prototype.forEach`, or using map to manipulate the array's elements and return a new array